src/Core/src/Platform/iOS/SemanticExtensions.cs

```diff
@@ -11,8 +11,33 @@ public static void UpdateSemantics(this UIView platformView, IView view)
 			if (semantics == null)
 				return;
 
-			platformView.AccessibilityLabel = semantics.Description;
-			platformView.AccessibilityHint = semantics.Hint;
+			var desc = semantics.Description;
+			var hint = semantics.Hint;
+
+			if (platformView is UISearchBar searchBar)
+			{
+				var textField = searchBar.GetSearchTextField();
+
+				if (textField == null)
+					return;
+				else
+					platformView = textField;
+
+				if (!string.IsNullOrWhiteSpace(desc))
+				{
+					textField.AccessibilityLabel = desc;
+				}
+
+				if (!string.IsNullOrWhiteSpace(hint))
+				{
+					textField.AccessibilityHint = hint;
+				}
```

Comment:
I think you can remove all of this code

And then move this code back outside the else

https://github.com/dotnet/maui/pull/7802/files#diff-49ae98b58eed27aaa20630cb71ec8bd192e3b3bf7a18e7fa66704b47c4a9b181R38-R39

Reply:
oops, yeah. removing the redundancy; thanks for the catch!